### PR TITLE
rgw_sal_motr: use GET for unversioned buckets

### PR DESCRIPTION
**Pull request by andriytk**
Use GET Motr index operation for reading unversioned buckets
instead of NEXT. GET is much more lightweight operation.

Solution: check the bucket at get_bucket_dir_ent(): if it's
unversioned one, use GET.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [x] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by andriytk** [comment]
Build version - https://eos-jenkins.colo.seagate.com/job/GitHub-custom-ci-builds/job/generic/job/custom-ci/6917/

**Comment by sjain09** [comment]
@andriytk, I don't see any change as per the description. No code modified related to index operation.
Am I missing something?

File: src/rgw/rgw_sal_motr.cc [hunk 1]
```diff
@@ -3061,7 +3062,7 @@ int MotrObject::get_bucket_dir_ent(const DoutPrefixProvider *dpp, rgw_bucket_dir
   if (this->get_key().have_null_instance())
     obj_key = this->get_name(); // drop "null" suffix
 
-  if (this->have_instance()) {
+  if (!bucket->get_info().versioned() || this->have_instance()) {
```
Comment (andriytk) [inline src/rgw/rgw_sal_motr.cc:3065]: @sjain09 ^ here is the change.

**Comment by andriytk** [comment]
> @andriytk, I don't see any change as per the description...

Sorry, it is easy to miss the 1-line-change indeed. Updated the description.

**Comment by shailesh-vaidya** [comment]
retest this please

**Comment by shailesh-vaidya** [comment]
retest this please

**Review by sjain09 — APPROVED** [review]

**Pull request closed without merging**